mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,161 @@
+    """
+    norm = (
+        np.linalg.norm(
+            cirq.unitary(opt)
+            - np.trace(cirq.unitary(opt))
+            / 2 ** int(np.log2(cirq.unitary(opt).shape[0])),
+            ord=np.inf,
+        )
+        ** 2
+    )
+    return float(norm)
+
+
+def n_measurements_opts_expectation_bound(
+    error: float,
+    observables: List[PauliString[Any]],
+    failure_rate: float,
+) -> Tuple[int, int]:
+    """
+    This function returns the minimum number of classical shadows required and
+    the number of groups "k" into which we need to split the shadows for
+    achieving the desired accuracy and failure rate in operator expectation
+    value estimation.
+
+    Args:
+        error: The error on the estimator.
+        observables: List of cirq.PauliString corresponding to the
+            observables we intend to measure.
+        failure_rate: Rate of failure for the bound to hold.
+
+    Returns:
+        Integers quantifying the number of snapshots required to satisfy
+        the shadow bound and the chunk size required to attain the specified
+        failure rate.
+    """
+    M = len(observables)
+    K = 2 * np.log(2 * M / failure_rate)
+
+    N = (
+        34
+        * max(local_clifford_shadow_norm(o) for o in observables)
+        / error**2
+    )
+    return int(np.ceil(N * K)), int(K)
+
+
+def fidelity(
+    state_vector: NDArray[np.complex64],
+    rho: NDArray[np.complex64],
+) -> float:
+    """
+    Calculate the fidelity between a state vector and a density matrix.
+    Args:
+        state_vector: The vector whose norm we want to calculate.
+        rho: The operator whose norm we want to calculate.
+
+    Returns:
+        Scalar corresponding to the fidelity.
+    """
+    return np.reshape(state_vector.conj().T @ rho @ state_vector, -1).real[0]
+
+
+def transform_to_cirq_paulistring(
+    pauli_str: Union[str, mitiq.PauliString, cirq.PauliString[Any]]
+) -> Tuple[float, cirq.PauliString[Any]]:
+    """Transforms mitiq.PauliString, cirq.PauliString, cirq.PauliSum or string
+    to a cirq.PauliString class.
+
+    Args:
+      pauli_str: A mitiq.PauliString, cirq.PauliString, cirq.PauliSum
+        or string.
+
+    Returns:
+      A tuple containing the coefficient and a cirq.PauliString.
+    """
+    if isinstance(pauli_str, cirq.PauliString):
+        # Extract coefficient and return cirq.PauliString without coefficient
+        coeff = np.real(pauli_str.coefficient)  # type: ignore
+        new_pauli_str: cirq.PauliString[Any] = cirq.PauliString(
+            {qubit: op for qubit, op in pauli_str.items()}
+        )
+        return coeff, new_pauli_str
+
+    elif isinstance(pauli_str, mitiq.PauliString):
+        qubit_pauli_map = {
+            cirq.LineQubit(i): gate
+            for i, gate in enumerate(pauli_str._pauli.values())
+        }
+        return np.real(pauli_str.coeff), cirq.PauliString(
+            qubit_pauli_map=qubit_pauli_map
+        )
```

Comment:
```suggestion
    elif isinstance(pauli_str, mitiq.PauliString):
        return pauli_str._pauli
```
`mitiq.PauliString._pauli` is a `cirq.PauliString` so maybe we can just return that.

Reply:
removed the whole function now.